ocaml/xapi/xapi_pif.ml

```diff
@@ -451,20 +451,17 @@ let introduce_internal
   pif
 
 (* Assertion passes if network has clusters attached but host has disabled clustering *)
-let assert_no_clustering_enabled ~__context ~network ~host =
+let assert_no_clustering_enabled ~__context ~host =
   if not (Xapi_clustering.is_clustering_disabled_on_host ~__context host)
-  then
-    (Db.Cluster.get_refs_where ~__context
-      ~expr:Db_filter_types.(Eq(Field "network", Literal (Ref.string_of network))))
-    |> function
-    | []   -> ()
-    | _::_ -> raise Api_errors.(Server_error (clustering_enabled_on_network, [Ref.string_of network]))
+  then begin match Xapi_clustering.find_cluster_host ~__context ~host with
+    | Some cluster_host ->
+      raise Api_errors.(Server_error (clustering_enabled, [Ref.string_of cluster_host]))
```

Comment:
Looks like this always fails if clustering is enabled, which matches the function's name but in this case the function's name is wrong.